### PR TITLE
feat(#7367): ProgressBar for `ExportAsJSONAction`

### LINKED ISSUE #7367: Export as JSON needs an activity modal
**Is your feature request related to a problem? Please describe.**
The "Export As JSON" action can take a long time to execute. Currently users are given no feedback that an export is taking place, so with large exports it can appear that the export has silently failed.

**Describe the solution you'd like**
A progress or activity overlay informing the user that an export is happening.

### LINKED ISSUE #7380: NASA moved the Example Imagery files
#### Summary

Example Imagery loads its images from NASA's image archives. These images now 404, likely due to NASA's recent work on web consolidation and migration to WordPress. These URLs likely just need to be updated.

#### Expected vs Current Behavior

The image urls should not 404

#### Steps to Reproduce

1. Load example imagery
2. Images no loady

#### Impact Check List

- [x] Data loss or misrepresented data?
- [x] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [ ] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [x] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

#### Additional Information
We reference images hosted by nasa hq like 'https://www.hq.nasa.gov/alsj/a16/AS16-117-18748.jpg' in our example imagery plugin. This changed to 'https://www.nasa.gov/wp-content/uploads/static/history/alsj/a16/AS16-117-18731.jpg'

### PR DESCRIPTION
**Pull request by ozyx**
Closes #7367 
Closes #7380 

### Describe your changes:

Adds a ProgressBar to asynchronous Export as JSON actions (such as with external persistence i.e. CouchDB).

Also a few refactors and minor fixes:

- Updates the URLs for example imagery
- Updates `LocalStorageProvider` to use the actual `localStorage` api (`getItem()` and `setItem()`, etc).
- Updates `ExportAsJSONAction` to use private methods
- Refactors out the `model` prop object in ProgressDialog-- use individual props instead

https://github.com/nasa/openmct/assets/9259993/a4d4ef59-1423-41de-bc95-a72cee737909

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Has this been smoke tested?
* [x] Have you associated this PR with a `type:` label? Note: this is not necessarily the same as the original issue.
* [x] Have you associated a milestone with this PR? Note: leave blank if unsure.
* [ ] Is this a breaking change to be called out in the release notes?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [x] Changes appear to address issue?
* [x] Reviewer has tested changes by following the provided instructions?
* [x] Changes appear not to be breaking changes?
* [x] Appropriate automated tests included?
* [x] Code style and in-line documentation are appropriate?

### CODE REVIEW

File: src/plugins/localStorage/LocalStorageObjectProvider.js [hunk 1]
```diff
@@ -71,14 +71,14 @@ export default class LocalStorageObjectProvider {
    * @private
    */
   persistSpace(space) {
-    this.localStorage[this.spaceKey] = JSON.stringify(space);
+    this.localStorage.setItem(this.spaceKey, JSON.stringify(space));
```
Comment (ozyx) [inline src/plugins/localStorage/LocalStorageObjectProvider.js:74]: this is a driveby, but figured we should be using the localStorage API here

File: e2e/tests/functional/plugins/plot/logPlot.e2e.spec.js [hunk 2]
```diff
@@ -116,14 +116,9 @@ async function makeOverlayPlot(page, myItemsFolderName) {
...
-  await page.locator('div:nth-child(5) .c-form-row__controls .form-control .field input').fill('6');
-
-  await page.locator('div:nth-child(6) .c-form-row__controls .form-control .field input').click();
-  await page.locator('div:nth-child(6) .c-form-row__controls .form-control .field input').fill('4');
-
-  await page.locator('div:nth-child(7) .c-form-row__controls .form-control .field input').click();
-  await page.locator('div:nth-child(7) .c-form-row__controls .form-control .field input').fill('2');
+  await page.getByLabel('Period').fill('2');
+  await page.getByLabel('Amplitude').fill('6');
+  await page.getByLabel('Offset').fill('4');
```
Comment (unlikelyzero) [inline e2e/tests/functional/plugins/plot/logPlot.e2e.spec.js:121]: lol

**Review by scottbell — APPROVED** [review]
Good looks good. Example imagery now working:
Screenshot 2024-01-18 at 12 11 45 PM

and JSON exports for CouchDB now have a nifty progress bar:

https://github.com/nasa/openmct/assets/9853862/52a924f3-8e7e-4306-b22e-9a73a7184b37
